pgraph/pgraph_test.go

```diff
@@ -627,6 +627,56 @@ func TestPgraphT11(t *testing.T) {
 	}
 }
 
+// AddEdge should be the dual of DeleteEdge.
+func TestPgraphDeleteEdge1(t *testing.T) {
+	G := NewGraph("g")
+	v1 := NV("v1")
+	v2 := NV("v2")
+	e1 := NewEdge("e1")
+	save := *G
+
+	G.AddEdge(v1, v2, e1)
+	G.DeleteEdge(e1)
+
+	runGraphCmp(t, G, &save)
+}
+
+// adding an edge should be the same after a delete
+func TestPgraphDeleteEdge2(t *testing.T) {
+	G := NewGraph("g")
+	v1 := NV("v1")
+	v2 := NV("v2")
+	e1 := NewEdge("e1")
+	save := *G
+
+	G.AddEdge(v1, v2, e1)
+	G.DeleteEdge(e1)
+	G.AddEdge(v1, v2, e1)
+
+	save.AddEdge(v1, v2, e1)
+
+	runGraphCmp(t, G, &save)
+}
+
+// adding a different edge should be the same after a delete
+func TestPgraphDeleteEdge3(t *testing.T) {
+	G := NewGraph("g")
+	v1 := NV("v1")
+	v2 := NV("v2")
+	v3 := NV("v3")
+	e1 := NewEdge("e1")
+	e2 := NewEdge("e2")
+	save := *G
+
+	G.AddEdge(v1, v2, e1)
+	G.AddEdge(v2, v3, e2)
+
+	G.DeleteEdge(e1)
```

Comment:
Here you change both `G` and `save`. I think this is a bit confusing! Please keep this ultra simple. Basic tests first test the core things in an extremely straight forward and boring way.

You should be creating `g1` and `g2` in all your tests. You can then perform some operations on `g2` so that it looks like `g1`. Then you run the cmp function on `g1` and `g2`. Isn't that simpler?